docs/monitoring.md

```diff
@@ -95,6 +95,49 @@ The history server can be configured as follows:
   </tr>
 </table>
 
+### Applying compaction on rolling event log files
+
+A long-running application (e.g. streaming) can bring a huge single event log file which may cost a lot to maintain and
+also requires a bunch of resource to replay per each update in Spark History Server.
+
+Enabling <code>spark.eventLog.rolling.enabled</code> and <code>spark.eventLog.rolling.maxFileSize</code> would
+let you have rolling event log files instead of single huge event log file which may help some scenarios on its own,
+but it still doesn't help you reducing the overall size of logs.
+
+Spark History Server can apply 'compaction' on the rolling event log files to reduce the overall size of
```

Comment:
Nit: Not sure what's the intention with this `‘` sign here. Maybe `compaction` word is not so special that it must be highlighted. This applies more places.

Reply:
That's used as emphasizing "what" we are explaining here, but maybe not a big deal. Will remove.